cime_config/tests.py

```diff
@@ -13,13 +13,21 @@
     "e3sm_mosart_developer" : {
         "share" : True,
         "time"  : "0:45:00",
+        "inherit" : ("e3sm_mosart_heat"),
```

Comment:
@liho745 Thanks for adding the test. Instead of creating a new test suite (i.e. `e3sm_mosart_heat`), please directly add the new test in the `e3sm_mosart_developer` test suite at https://github.com/E3SM-Project/E3SM/pull/5343/files#diff-e2fbf220fd98012c3bdf61d7f341a1815f43823af2ffd3bf926d5aafc4363d8bR20. Does it make sense?

Reply:
I will have to revisit the approach taken for the sediment test when I re-merge that PR into `next`. To avoid creating many new test suites, directly add `ERS.r05_r05.RMOSGPCC.mosart-heat` in the `e3sm_mosart_developer` test suite.

Reply:
>I quote from him "mosart-sediment cannot run with other test cases with the same build".

Okay, I now recall this issue with mosart sediment. Does your mosart heat test need different build flags (not namelist flags) than those tests in the e3sm_mosart_developer test suite?

Reply:
Yes, I have to turn on heatflag via user_nl_mosart.

Reply:
That is a namelist flag and not a build flag (e.g. `-DUSE_PETSC_LIB`), which is used when compiling various `*.F90`. Thus, you should be able to directly add the new test in the e3sm_mosart_developer test suite.

Reply:
> That is a namelist flag and not a build flag (e.g. `-DUSE_PETSC_LIB`), which is used when compiling various `*.F90`. Thus, you should be able to directly add the new test in the e3sm_mosart_developer test suite.

I tried what you said, and the test results are below, i.e., the baseline test failed. 
====================
  ERS.r05_r05.RMOSGPCC.compy_intel.mosart-heat (Overall: DIFF) details:
    PASS ERS.r05_r05.RMOSGPCC.compy_intel.mosart-heat CREATE_NEWCASE
    PASS ERS.r05_r05.RMOSGPCC.compy_intel.mosart-heat XML
    PASS ERS.r05_r05.RMOSGPCC.compy_intel.mosart-heat SETUP
    PASS ERS.r05_r05.RMOSGPCC.compy_intel.mosart-heat SHAREDLIB_BUILD time=277
    FAIL ERS.r05_r05.RMOSGPCC.compy_intel.mosart-heat NLCOMP
    PASS ERS.r05_r05.RMOSGPCC.compy_intel.mosart-heat MODEL_BUILD time=212
    PASS ERS.r05_r05.RMOSGPCC.compy_intel.mosart-heat SUBMIT
    PASS ERS.r05_r05.RMOSGPCC.compy_intel.mosart-heat RUN time=51
    PASS ERS.r05_r05.RMOSGPCC.compy_intel.mosart-heat COMPARE_base_rest
    FAIL ERS.r05_r05.RMOSGPCC.compy_intel.mosart-heat BASELINE 21ffb4d: ERROR BFAIL baseline directory '/compyfs/liho623/e3sm_scratch/e3sm_baseline/21ffb4d/ERS.r05_r05.RMOSGPCC.compy_intel.mosart-heat' does not exist
    PASS ERS.r05_r05.RMOSGPCC.compy_intel.mosart-heat MEMLEAK
    PASS ERS.r05_r05.RMOSGPCC.compy_intel.mosart-heat SHORT_TERM_ARCHIVER

Reply:
The above failure is okay because the baseline for this new test doesn't exist.

Reply:
Sounds good. Added the mosart-heat test directly into the e3sm_mosart_developer suite.